### PR TITLE
DEV: Deleted the "ChatComposerPresenceManager"...

### PR DESCRIPTION
**Pull request by ZogStriP**
...since it was mostly duplicating the work the "ComposerPresenceManager" was doing.

So now the #chat composer uses the same "presence manager" as the composer, benefiting from the "hide presence" checks, with the only difference that the "keep alive" timeout is 5s for chat and 10s for topics/posts.

Extracted from #30067

### CODE REVIEW

File: plugins/discourse-presence/assets/javascripts/discourse/services/composer-presence-manager.js [hunk 1]
```diff
@@ -2,63 +2,45 @@ import { cancel, debounce } from "@ember/runloop";
 import Service, { service } from "@ember/service";
 import { isTesting } from "discourse-common/config/environment";
 
-const PRESENCE_CHANNEL_PREFIX = "/discourse-presence";
-const KEEP_ALIVE_DURATION_SECONDS = 10;
+const KEEP_ALIVE = 10 * 1000; // 10 seconds
 
 export default class ComposerPresenceManager extends Service {
   @service presence;
```
Comment (CvX) [inline plugins/discourse-presence/assets/javascripts/discourse/services/composer-presence-manager.js:8]: while we're here:

```suggestion
  @service currentUser;
  @service presence;
  @service siteSettings;
```

File: plugins/discourse-presence/assets/javascripts/discourse/services/composer-presence-manager.js [hunk 2]
```diff
@@ -2,63 +2,45 @@ import { cancel, debounce } from "@ember/runloop";
...
 
-  willDestroy() {
-    this.leave();
-  }
```
Comment (CvX) [inline plugins/discourse-presence/assets/javascripts/discourse/services/composer-presence-manager.js:13]: the idea is that components using this manager are calling the leave on their `willDestroy` anyway?

Comment (ZogStriP) [reply plugins/discourse-presence/assets/javascripts/discourse/services/composer-presence-manager.js:13]: I added a `debugger` in there and it was never called. Even though I navigated everywhere in the app so I decided to remove it 🤷‍♂️

Comment (CvX) [reply plugins/discourse-presence/assets/javascripts/discourse/services/composer-presence-manager.js:13]: yeah, it wouldn't trigger in the normal app usage - it's for the test env where services are destroyed with the application instance between tests, but as long the service doesn't leak state outside itself/app instance there's no need for special cleanup

**Review by CvX — APPROVED** [review]

**Pull request merged**